gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -175,9 +176,19 @@ public static ClientContext create(StubSettings settings) throws IOException {
     // A valid EndpointContext should have been created in the StubSettings
     EndpointContext endpointContext = settings.getEndpointContext();
     String endpoint = endpointContext.resolvedEndpoint();
+    String apiKey = settings.getApiKey();
+    Credentials credentials = settings.getCredentialsProvider().getCredentials();
+    if (apiKey != null && credentials != null) {
```

Comment:
I added this check here instead of in the build() method on ClientSettings, as it made the code more readable on line 187 to me - we're not overriding client credentials, as they would have to be null.